### PR TITLE
Sync Process Instance Migration gRPC documentation with gateway.proto

### LINKED ISSUE #15568: Sync Process Instance Migration gRPC documentation with gateway.proto
### Description

Response codes and their explanations are out of sync with what we documented on our Camunda 8 gRPC API docs. The issue is created to fix that problem and have the same error code explanations in both places:

`gateway.proto`: https://github.com/camunda/zeebe/blob/main/gateway-protocol/src/main/proto/gateway.proto
gRPC doc: https://github.com/camunda/camunda-docs/pull/2955/files/b7d1ef9a7803044e815d5bd93bc849815b74bdc0#diff-d8972d11490d2d55a8153807a50d23665ce723dce9c70e7b52ba27ba0627f53cR821

Related change request: https://github.com/camunda/camunda-docs/pull/2955#discussion_r1420217502

### PR DESCRIPTION
**Pull request by berkaycanbc**
## Description

Process Instance Migration gRPC error codes are updated. Explanations of codes simply aligned with what we added in https://github.com/camunda/camunda-docs/blob/eb99438b28d5b3b0528cf9422cdfac3db8df3ad9/docs/apis-tools/grpc.md#grpc_status_not_found-6

## Related issues

closes #15568 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway-protocol/src/main/proto/gateway.proto [hunk 1]
```diff
@@ -919,19 +919,21 @@ service Gateway {
 
     Errors:
       NOT_FOUND:
-        - the process instance referenced by `processInstanceKey` is not active
+        - no process instance exists with the given key, or it is not active
+        - no process definition exists with the given target definition key
```
Comment (koevskinikola) [inline gateway-protocol/src/main/proto/gateway.proto:923]: :question: Should we mention tenants here? We do it in some other commands, so we would be aligned.

Comment (berkaycanbc) [reply gateway-protocol/src/main/proto/gateway.proto:923]: @koevskinikola Yes, we should update multi tenancy behaviour here as well as in the documentation. But at the moment, it is not decided what do with multi tenancy. Let's wait for the result and I will update here accordingly.

Also, rejection cases for events (see #15405 and #15406) needs to be updated.

**Comment by berkaycanbc** [comment]
I sent a small update to mention rejection cases for events (see #15405 and #15406).

**Comment by berkaycanbc** [comment]
@koevskinikola I added the NOT_FOUND case for multi-tenancy, please take a look. 👀

**Review by koevskinikola — APPROVED** [review]
LGTM @berkaycanbc! :rocket: 

bors merge

**Pull request merged**